src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/betti_numbers.jl

```diff
@@ -34,7 +34,7 @@ function betti_number(v::NormalToricVarietyType, i::Int)
     end
     
     # extract vector of currently-known Betti numbers (or create it if necessary)
-    betti_numbers = get_attribute!(() -> fill(ZZRingElem(-1), d+1), v, :betti_number)::Vector{ZZRingElem}
+    betti_numbers = get_attribute!(v, :betti_number, [ZZ(-1) for _ in 1:(d+1)])::Vector{ZZRingElem}
```

Comment:
This always allocates `[ZZ(-1) for _ in 1:(d+1)]` and if the attribute is already set, discards it.

Reply:
```suggestion
    betti_numbers = get_attribute!(() -> [ZZ(-1) for _ in 1:(d+1)], v, :betti_number)::Vector{ZZRingElem}
```
True. So let's revert this (up to the aliasing issue)